swap/src/config.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Result;
+use libp2p::multiaddr::Multiaddr;
+use serde::{Deserialize, Serialize};
+use std::{fs, path::Path};
+use url::Url;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Config {
+    pub bitcoind_url: Url,
+    pub listen_addr: Multiaddr,
+}
+
+impl Config {
+    pub fn load(path: &Path) -> Result<Config> {
+        let contents = fs::read_to_string(path).expect("Something went wrong reading the file");
+        let file: Config = toml::from_str(contents.as_str())?;
+        Ok(file)
+    }
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            bitcoind_url: Url::parse("https://127.0.0.1:8332")
+                .expect("Failed to generate default config"),
```

Comment:
```suggestion
                .expect("Failed to parse static string"),
```

Reply:
something like "parse static string" is printed anyway if parse fails. The string in expect() is providing context